### PR TITLE
Edit Get & Set ShortTermRetentionPolicy For Adding New Parameter: DiffBackupIntervalInHours

### PR DESCRIPTION
**Pull request by lululilliancoding**
## Description

ShortTermRetentionPolicy Get and Set cmdlets to include new parameter DiffBackupIntervalInHours. 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by isra-fel** [comment]
Adding "DO NOT MERGE" since SDK is not ready

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs [hunk 1]
```diff
@@ -34,6 +34,14 @@ public class SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy : AzureSqlDatab
         [ValidateNotNullOrEmpty]
         public int RetentionDays{ get; set; }
 
+        /// <summary>
+        /// Gets or sets differential backup interval hours.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 4,
```
Comment (isra-fel) [inline src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs:41]: Too many positional parameters can make the command difficult to use. Please remove this line.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs [hunk 2]
```diff
@@ -34,6 +34,14 @@ public class SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy : AzureSqlDatab
...
+            HelpMessage = "The differential backup interval, in hours.")]
+        public int? DiffBackupIntervalInHours { get; set; }
```
Comment (isra-fel) [inline src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs:43]: Please use an *optional* `int` parameter instead of a *mandatory* `int?` parameter.

Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs:43]: But we accept DiffBackupIntervalInHours == null.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs (hunk 1 above)
Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs:41]: OK will update in next push.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs (hunk 2 above)
Comment (isra-fel) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs:43]: "Nullable" is a concept in .NET. In PowerShell, it is represented by the existance / absence of a parameter.

If SDK accepts a `int?`, you should write this in your code:
```csharp
this.managementClient.someOperation(new SomeModel() {
    DiffBackupIntervalInHours = this.IsParameterBound(c => c.DiffBackupIntervalInHours) ? DiffBackupIntervalInHours : null as int?
})

Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs:43]: where do I put this to?

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs (hunk 1 above)
Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs]: removed.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs [hunk 3]
```diff
@@ -29,11 +29,16 @@ public class SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy : AzureSqlDatab
...
-        [ValidateNotNullOrEmpty]
-        public int RetentionDays{ get; set; }
+        public int? RetentionDays { get; set; }
 
+        /// <summary>
+        /// Gets or sets differential backup interval hours.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The differential backup interval, in hours.")]
+        public int? DiffBackupIntervalInHours { get; set; }
```
Comment (isra-fel) [inline src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs:40]: (1/2) make the parameter optional, and do not use nullable types
This will make the parameter easier to undertand and use
```suggestion
        [Parameter(Mandatory = false,
            HelpMessage = "The differential backup interval, in hours.")]
        public int DiffBackupIntervalInHours { get; set; }	
```

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs [hunk 4]
```diff
@@ -64,7 +69,7 @@ protected override IEnumerable<AzureSqlDatabaseBackupShortTermRetentionPolicyMod
                     ResourceGroupName,
                     ServerName,
                     DatabaseName,
-                    new Management.Sql.Models.BackupShortTermRetentionPolicy(retentionDays: RetentionDays))
+                    new Management.Sql.Models.BackupShortTermRetentionPolicy(retentionDays: RetentionDays, diffBackupIntervalInHours: DiffBackupIntervalInHours))
```
Comment (isra-fel) [inline src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs:72]: (2/2) convert `int` to `int?` then pass it to management SDK
```suggestion
                    new Management.Sql.Models.BackupShortTermRetentionPolicy(retentionDays: RetentionDays, diffBackupIntervalInHours: this.IsParameterBound(c => c.DiffBackupIntervalInHours) ? DiffBackupIntervalInHours : null as int?))
```

**Review by isra-fel — CHANGES_REQUESTED** [review]
To be specific, what I suggest are two changes to all the nullable parameters.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs (hunk 3 above)
Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs]: Modified.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs (hunk 4 above)
Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs]: Modified. For parameter retentionDays and diffBackupInterval: 
1. When they are both null -> throw "Parameter 'RetentionDays' and 'DiffBackupIntervalInHours' both being null is not allowed." error -> we expect this -> done. 
However, 
2. When one parameter is null and the other one has value -> it also throw " Nullable object must have a value." error -> we don't expect this since we want the configured parameter can go to server and been updated. The server can ignore the other parameter.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupShortTermRetentionPolicy.cs [hunk 5]
```diff
@@ -28,12 +29,17 @@ public class SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy : AzureSqlDatab
         /// <summary>
         /// Gets or sets backup retention days.
         /// </summary>
-        [Parameter(Mandatory = true,
-            Position = 3,
```
Comment (isra-fel) [inline src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupShortTermRetentionPolicy.cs:32]: I believe the breaking changes about parameter sets are caused by the removal of `Position = 3`.
Changing a positional parameter to a normal parameter is a breaking change -- user may omit parameter name in their script, but now it's not allowed

Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupShortTermRetentionPolicy.cs:32]: So it doesn't matter which position it is since parameter value always after parameter name. 
How can we skip these breaking changes in this PR? Looks like there is no real breaking changes.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs (hunk 4 above)
Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy.cs]: I modified as I guessed and it works perfect now. Either retentionDays or diffBackupInterval can be set individually.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupShortTermRetentionPolicy.cs (hunk 5 above)
Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupShortTermRetentionPolicy.cs:32]: @isra-fel

**Review by isra-fel — APPROVED** [review]
I undid the position of `DiffBackupIntervalInHours` so that static analysis won't complain about "too many positional parameters".
I've also regenerated help docs so that the diff of `src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md` makes more sense.

**Comment by lululilliancoding** [comment]
> I undid the position of `DiffBackupIntervalInHours` so that static analysis won't complain about "too many positional parameters".
> I've also regenerated help docs so that the diff of `src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md` makes more sense.

That's making sense. Thanks a lot!

**Comment by lululilliancoding** [comment]
The pipeline hit "run more than 180 minutes" error. I just clicked re-run.

File: src/Sql/Sql/ChangeLog.md [hunk 6]
```diff
@@ -18,41 +18,7 @@
...
-* Added cmdlet `Get-AzSqlInstanceActiveDirectoryOnlyAuthentication`
-* Added cmdlet `Enable-AzSqlInstanceActiveDirectoryOnlyAuthentication`
-* Added cmdlet `Disable-AzSqlInstanceActiveDirectoryOnlyAuthentication`
-* Updated cmdlets `New-AzSqlDatabaseImport` and `New-AzSqlDatabaseExport` to support network isolation functionality
-* Added cmdlet `New-AzSqlDatabaseImportExisting`
-* Updated Databases cmdlets to support backup storage type specification
-* Added Force parameter to `New-AzSqlDatabase`
-* Updated ActiveDirectoryOnlyAuthentication cmdlets for server and instance to include ResourceId and InputObject
-* Added support for Managed HSM Uris for SQL DB and Managed Instance
+* Added DiffBackupIntervalInHours to `Set-AzSqlDatabaseBackupShortTermRetentionPolicy`
```
Comment (isra-fel) [inline src/Sql/Sql/ChangeLog.md:21]: I just noticed that the change logs of past versions were deleted in this PR. Please be careful when you submit another PR. Thanks
